components/brave_rewards/browser/rewards_notification_service_impl.cc

```diff
@@ -91,11 +91,15 @@ void RewardsNotificationServiceImpl::DeleteNotification(
     rewards_notification = rewards_notifications_[id];
     rewards_notifications_.erase(id);
   }
+  StoreRewardsNotifications();
+
   OnNotificationDeleted(rewards_notification);
 }
 
 void RewardsNotificationServiceImpl::DeleteAllNotifications() {
   rewards_notifications_.clear();
+  rewards_notifications_displayed_.clear();
+  StoreRewardsNotifications();
```

Comment:
why this change?

Reply:
Why we clear vector or again store an empty vector because before it was stored on destructor only?

Reply:
I restored it in destructor. @NejcZdovc  @SergeyZhukovsky is it enough to resolve this?

Reply:
same as above, if we need this it should be handled with an observer for `OnAllNotificationsDeleted`